### DISCUSSION TITLE
adding subscripts in expressions

### DISCUSSION THREAD

**Discussion by jchatkinson** [josdejong/mathjs#2590] [Q&A] [answered]
I'm working on a React component which uses mathjs expressions to both pretty print tex and evaluate the solution. An example of implemented component is linked below. 

[https://calcs.app/analysis-shape-properties](https://calcs.app/analysis-shape-properties)

In that example, the area formula is both tex'ified and evaluated from the string `A = b*d`

This works great, but I am wondering if it is possible to add subscripts to the variables while still maintaining the ability to both generate tex and evaluate the expression?

For instance, I might want to call the area `A_{rect} = b*d`

I don't see any mention of subscripts in the docs, and trying the obvious formats doesn't appear to work

**Comment by josdejong** [comment, marked as answer]
For variables names you can use characters like letters, numbers (except as first character), and underscrore `_`. However the brackets `{` and `}` have a special meaning: these are used to create an object like `A = {"name": "Joe"}`. So you can't use these characters in a variable name.

If you do need something like that, you could maybe do some post processing of the names yourself, like in expressions you use `A_rect`, and for displaying in tex you replace the part after the underscore with `A_{rect}`. To achieve that, you can write your own LaTeX handler: https://mathjs.org/docs/expressions/customization.html#handler

**Reply by jchatkinson** [reply]
Okay wow that was easier than I thought it would be. Its nice how extensible the expressions module was made. Thanks for pointing me in the right direction. 

for sake of closure in case anyone else needs to do something similar, my handler is shown below
```js
// create a custom handler to allow tex printing of subscripts in variable names (eg. A_rect => A_{rect} )
// https://mathjs.org/docs/expressions/customization.html#handler
const subscriptHandler = (node, options) => {
  if (node.type === 'SymbolNode' && node.name.includes('_')) {
    const index = node.name.indexOf('_');
    return ` ${node.name.substring(0,index+1)}{${node.name.substring(index+1)}}`
  }
}
...
...
// the handler can be activated by passing it to toTex() like so
texStr = eqn.toTex({handler: subscriptHandler})
```

**Reply by josdejong** [reply]
Good to hear, thanks for sharing your solution 👍
